src/transform/src/canonicalize_mfp.rs

```diff
@@ -285,6 +315,92 @@ impl CanonicalizeMfp {
         }
     }
 
+    /// 1. Removes such OR args in which there are contradicting literal constraints.
+    /// 2. Also, if an OR arg doesn't have any contradiction, this fn just deduplicates the
+    /// AND arg list of that OR arg.
+    ///
+    /// Example for 1:
+    /// `<arg1> OR (a = 5 AND a = 5 AND a = 8) OR <arg3>`
+    /// -->
+    /// `<arg1> OR <arg3> `
+    ///
+    /// Example for 2:
+    /// `<arg1> OR (a = 5 AND a = 5 AND b = 8) OR <arg3>`
+    /// -->
+    /// `<arg1> OR (a = 5 AND b = 8) OR <arg3>`
+    fn remove_impossible_or_args(mfp: &mut MapFilterProject) -> Result<bool, RecursionLimitError> {
+        let mut or_args = Self::get_or_args(mfp);
+        if or_args.len() == 0 {
+            return Ok(false);
+        }
+        let mut to_remove = Vec::new();
+        let mut changed = false;
+        or_args.iter_mut().enumerate().for_each(|(i, or_arg)| {
+            if let MirScalarExpr::CallVariadic {
+                func: VariadicFunc::And,
+                exprs: and_args,
+            } = or_arg
+            {
+                if Self::has_duplicates(and_args.clone()) {
+                    changed = true;
+                    and_args.sort();
+                    and_args.dedup();
+                }
```

Comment:
You might as well sort and dedup always and then compare the before to the after.